[JENKINS-63002] Revamp tabs and bigtables

fqueiruga opened the pull request:
See [JENKINS-63002](https://issues.jenkins-ci.org/browse/JENKINS-63002).

This PR revamps the tables and tab styles according to the design briefs provided on the Jira ticket. The scope of this PR is limited to tables implementing the `.bigtable` class. 

**NOTE:** the design was done with the assumption that we want to keep the tables and tabs together as was decided on https://github.com/jenkinsci/jenkins/pull/1344

<details>
<summary><b>Screenshots</b></summary>

<img width="1707" alt="Captura de pantalla 2020-07-09 a las 15 14 56" src="https://user-images.githubusercontent.com/5738588/87044607-131d6d80-c1f7-11ea-95f2-7a1a0548d7cd.png">

<img width="1707" alt="Captura de pantalla 2020-07-09 a las 15 15 04" src="https://user-images.githubusercontent.com/5738588/87044616-16185e00-c1f7-11ea-8a76-481a2f677367.png">

<img width="1707" alt="Captura de pantalla 2020-07-09 a las 15 15 09" src="https://user-images.githubusercontent.com/5738588/87044621-17498b00-c1f7-11ea-8516-cab00be43123.png">
</details>

#### Notable changes

- The tables and tabs have updated colors that are consistent with the new Jenkins color palette. They are consistent with sidebar and footer. They also have improved in-cell spacing.
- The components are fully themeable.
- Tables are no longer supposed to be wrapped in a `.pane-frame` class. This generated some inconsistencies. Tables inside a `.pane-frame` will not look bad, but should gradually be migrated. Only a [few plugins](https://github.com/search?l=XML&q=org%3Ajenkinsci+pane-frame+-repo%3Ajenkinsci%2Fjenkins+-path%3Atarget&type=Code) are affected.
- Tables and tabs CSS code is refactored into LESS modules.
- The `tabBar.jelly` widget now admits a `showBaseline` parameter that will render the underline when set to `"true"`.


#### Follow up work

There should be follow-up PR's to iron out the `.pane-frame` appearance, especially on the plugins management _Advanced_ page. It's also pending to update plugins by remove the `.pane-frame` wrapper.


### Proposed changelog entries

* Entry 1: JENKINS-63002, Revamped the tabs and tables with better spacing and a consistent color palette.

<!-- Comment: 
The changelogs will be integrated by the core maintainers after the merge.  See the changelog examples here: https://jenkins.io/changelog/ -->

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

<!-- For new API and extension points: Link to the reference implementation in open-source (or example in Javadoc) -->

### Desired reviewers

@timja
@uhafner 
@daniel-beck 
@oleg-nenashev 

<!-- Comment:
If you need an accelerated review process by the community (e.g., for critical bugs), mention @jenkinsci/code-reviewers
-->

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

fqueiruga committed:
Extracts tab, panes, tables styles to a module

fqueiruga committed:
Extract tab & table variables

fqueiruga committed:
Unwraps view and plugin tables from .pane-frame

fqueiruga committed:
Update table styles

- Styles are applied to the .bigtable class. There are some changes to .pane-frame
- Some styles are inherited from .pane-frame

fqueiruga committed:
Fix widget styles conflicting with tables

- Differentiates the side-panel widget pane borders from .pane-frame elements
- Fix padding spaces on project view header. The Functions#nbspIndent function was changed to return 2 less spaces
- Restored missing border on filtered plugins management table
- Fixed the alignment of some tables

fqueiruga committed:
Restyles the tab baseline and new item button

- The tabBar.jelly now has a prop to show a baseline. The default display of the baseline can also be themed

fqueiruga commented:
### TODO

- [ ] Upload screenshots.
- [ ] Test all `.pane-frame` uses, including the no-job page.
- [ ] Validate the CSS vars API.
- [ ] Verify that it works on IE 11.

timja commented on core/src/main/java/hudson/Functions.java:
```diff
@@ -755,7 +755,7 @@ public static String nbspIndent(String size) {
         i = Integer.parseInt(i > 0 ? size.substring(0, i) : size) / 10;
         StringBuilder buf = new StringBuilder(30);
-        for (int j = 0; j < i; j++)
+        for (int j = 2; j <= i; j++)
```
do I even want to know?

fqueiruga replied:
You were too fast, I was about to comment on this hack to ask for advice. This function is use to pad the icon column names so that the letter is positioned above the icon. It's a weird one that I did not know how to fix:

Old:
<img width="111" alt="Captura de pantalla 2020-07-08 a las 9 53 54" src="https://user-images.githubusercontent.com/5738588/86892840-12a2ab00-c101-11ea-92a9-f218fa6e17ba.png">

New:
<img width="156" alt="Captura de pantalla 2020-07-08 a las 9 55 06" src="https://user-images.githubusercontent.com/5738588/86892890-25b57b00-c101-11ea-87de-d2598f349764.png">

I want to at least put a comment before it is merged.

fqueiruga commented on war/src/main/less/abstracts/theme.less:
```diff
@@ -151,23 +160,38 @@
+  --tab-link-text-color: var(--text-color-secondary);
+  --tab-link-bg: transparent;
+  --tab-link-border-color: var(--light-grey);
+  --tab-link-text-color--hover: var(--tab-link-text-color); // Redundant?
+  --tab-link-bg--hover: var(--light-grey);
+  --tab-link-border-color--hover: var(--tab-link-bg--hover); // Redundant?
+  --tab-link-text-color--active: var(--tab-link-text-color); // Redundant?
+  --tab-link-bg--active: var(--light-grey);
+  --tab-link-border-color--active: var(--tab-link-checked-bg); // Redundant?
```
@timja I think I might have gone too hard on the CSS var API. WDYT?

timja replied:
it's a lot, but seems quite good.

timja commented on war/src/main/less/abstracts/theme.less:
```diff
@@ -151,23 +160,38 @@
+  // TabBar links
+  --tab-link-border-radius: 4px;
+  --tab-link-border-width: 2px;
+  --tab-link-padding-x: 0.75rem;
```
do the paddings need to be exposed in the API?

fqueiruga replied:
I put them there because it made it easier for me to iterate the styles. Do you think they should be pruned?

fqueiruga replied:
I think we need to decide how far we want to go with the CSS vars

fqueiruga commented on war/src/main/less/abstracts/theme.less:
```diff
@@ -125,11 +128,17 @@
-  --table-parameters-bg-color--hover: var(--very-light-grey);
-  --table-stripped-bg-color--hover: #e8e8e8;
+  --even-row-color: var(--lightest-grey);
+  --bigtable-border-width: var(--pane-border-width);
+  --bigtable-header-bg: var(--pane-header-bg);
+  --bigtable-header-font-weight: bold; // Does specifying this make sense
+  --bigtable-header-font-size: var(--font-size-md);
+  --bigtable-header-text-color: var(--pane-header-text-color);
+  --bigtable-row-border-color: var(--medium-grey);
+  --bigtable-cell-padding-x: 0.75rem;
+  --bigtable-cell-padding-y: 0.5rem;
+  --table-parameters-bg--hover: var(--light-grey);
+  --table-stripped-bg--hover: var(--light-grey);
```
@timja do you agree with this var namespace `--bigtable` instead `--table`?

timja replied:
haven't looked as closely as you but why do we need to differentiate tables and bigtables?

Is it because tables are used for layout in many places and big table is actual tabular data?

fqueiruga replied:
`--table-stripped-bg--hover` is used here at least:
<img width="1409" alt="Captura de pantalla 2020-07-08 a las 10 36 36" src="https://user-images.githubusercontent.com/5738588/86897397-a5464880-c107-11ea-982b-0c0ca529400f.png">

Could be changed. I'm open to feedback.

fqueiruga replied:
Also, I'm not really sure about inheriting `--pane-*` styles. It feels right but I don't know how well it scales.

timja replied:
should this be striped not stripped?

fqueiruga replied:
Probably yes. It was an existing variable. Changing it.

fqueiruga commented:
@uhafner @timja I have played around with different header combinations. I'm also playing around with different border options.

### Separated tabs

<details>
<summary>Light grey header</summary>

<img width="1707" alt="Captura de pantalla 2020-07-09 a las 12 01 23" src="https://user-images.githubusercontent.com/5738588/87028431-e3fb0200-c1de-11ea-9455-1b93e1c0f2b0.png">

<img width="1707" alt="Captura de pantalla 2020-07-09 a las 12 01 37" src="https://user-images.githubusercontent.com/5738588/87028444-e78e8900-c1de-11ea-914f-bca1c44c4574.png">

<img width="1707" alt="Captura de pantalla 2020-07-09 a las 12 01 45" src="https://user-images.githubusercontent.com/5738588/87028448-e8bfb600-c1de-11ea-88a3-4fea57608077.png">

<img width="1707" alt="Captura de pantalla 2020-07-09 a las 12 01 50" src="https://user-images.githubusercontent.com/5738588/87028469-ee1d0080-c1de-11ea-9a1e-e2b3099e54cb.png">

<img width="1707" alt="Captura de pantalla 2020-07-09 a las 12 19 57" src="https://user-images.githubusercontent.com/5738588/87028500-f8d79580-c1de-11ea-841b-eb18171abcec.png">

<img width="1707" alt="Captura de pantalla 2020-07-09 a las 12 20 15" src="https://user-images.githubusercontent.com/5738588/87028513-fc6b1c80-c1de-11ea-9fd8-653510cffe1a.png">
</details>


<details>
<summary>White header (stripped rows will only look right after refactoring table headers using thead and tbody</summary>

<img width="1707" alt="Captura de pantalla 2020-07-09 a las 12 05 59" src="https://user-images.githubusercontent.com/5738588/87028745-5ec41d00-c1df-11ea-8a00-8769c6249195.png">

<img width="1707" alt="Captura de pantalla 2020-07-09 a las 12 06 50" src="https://user-images.githubusercontent.com/5738588/87028755-6388d100-c1df-11ea-97d5-c849524aced3.png">

<img width="1707" alt="Captura de pantalla 2020-07-09 a las 12 07 09" src="https://user-images.githubusercontent.com/5738588/87028757-64216780-c1df-11ea-9070-db690c20c6c0.png">

<img width="1707" alt="Captura de pantalla 2020-07-09 a las 12 07 45" src="https://user-images.githubusercontent.com/5738588/87028758-64b9fe00-c1df-11ea-992a-4a26327a1497.png">

<img width="1707" alt="Captura de pantalla 2020-07-09 a las 12 08 04" src="https://user-images.githubusercontent.com/5738588/87028763-671c5800-c1df-11ea-9e60-49b46eb2fd8c.png">

<img width="1707" alt="Captura de pantalla 2020-07-09 a las 12 12 22" src="https://user-images.githubusercontent.com/5738588/87028765-67b4ee80-c1df-11ea-8b8d-3369a499d3b1.png">

<img width="1707" alt="Captura de pantalla 2020-07-09 a las 12 18 10" src="https://user-images.githubusercontent.com/5738588/87028769-67b4ee80-c1df-11ea-9690-c5fe1bbe314f.png">

</details>


### Integrated tabs

<details>
<summary>Light grey header</summary>

<img width="1707" alt="Captura de pantalla 2020-07-09 a las 12 27 55" src="https://user-images.githubusercontent.com/5738588/87029129-fe81ab00-c1df-11ea-8dfb-869a1f258bc9.png">

<img width="1707" alt="Captura de pantalla 2020-07-09 a las 12 28 07" src="https://user-images.githubusercontent.com/5738588/87029146-02153200-c1e0-11ea-8ca7-445aac1b9b7e.png">

<img width="1707" alt="Captura de pantalla 2020-07-09 a las 12 28 56" src="https://user-images.githubusercontent.com/5738588/87029150-02adc880-c1e0-11ea-841f-e93e79a5a1e9.png">

<img width="1707" alt="Captura de pantalla 2020-07-09 a las 12 29 03" src="https://user-images.githubusercontent.com/5738588/87029153-03465f00-c1e0-11ea-84a5-65fe1480db40.png">

</details>


<details>
<summary>White header</summary>

<img width="1707" alt="Captura de pantalla 2020-07-09 a las 12 29 31" src="https://user-images.githubusercontent.com/5738588/87029204-15280200-c1e0-11ea-925b-0db2b1f63e97.png">

<img width="1707" alt="Captura de pantalla 2020-07-09 a las 12 29 47" src="https://user-images.githubusercontent.com/5738588/87029214-18bb8900-c1e0-11ea-984a-c74ba9a8dd55.png">

</details>


### My personal opinion

IMO the white background headers are not different enough. I have concerns that just changing the font-weight of the header is not an accessible option to identify these items.

timja commented:
Out of the screenshots I think the light grey one looks the best right now.

But I think that the white separated looks the best on the system information page and users page, possibly it could be best overall with the work on the pages that have the tabs on them

daniel-beck commented:
Integrated white but with lighter grey borders (like separate light grey), and a more pronounced divider to regular table rows ❤️

fqueiruga commented:
We could could add a prop to the tabBar widget to be able to choose separate and joined versions of the tabBar. Also, do you believe that the light grey header still looks good with white background tabs?

I think the white header stripped rows will only really work with an update for table markup to use `<thead>`, `<tbody>` and `<tfooter>`, but that may be difficult to pull off with some plugins.

> But I think that the white separated looks the best on the system information page and users page, possibly it could be best overall with the work on the pages that have the tabs on them

There could definitely be some variants, a bit out of scope for this PR though.



> Integrated white but with lighter grey borders (like separate light grey), and a more pronounced divider to regular table rows ❤️

Glad you like it. What do you mean with more pronounced dividers?

<details>
<summary>Btw, took some screenshots with a thinner, 1px medium grey border. Looks really good to me IMO</summary>

<img width="1707" alt="Captura de pantalla 2020-07-09 a las 12 49 46" src="https://user-images.githubusercontent.com/5738588/87031531-8b7a3380-c1e3-11ea-8c9f-f95490249388.png">

<img width="1707" alt="Captura de pantalla 2020-07-09 a las 12 50 11" src="https://user-images.githubusercontent.com/5738588/87031542-8f0dba80-c1e3-11ea-8604-3636dd32888f.png">

</summary>

fqueiruga commented:
Also, which of this proposed changes should go into this PR and which ones can be done on a follow-up one? I'd rather have some table updates by the next LTS than wait for the perfect combination.

daniel-beck commented:
> What do you mean with more pronounced dividers?

Make the separator between the title row and the rest of the rows stand out more. Right now it stands out far less than the top border / integrated tabs separator.

fqueiruga commented:
> Make the separator between the title row and the rest of the rows stand out more. Right now it stands out far less than the top border / integrated tabs separator.

I'll try to come up with something. The main issue will be the "Manage nodes and clouds" table, because it has table heading cells on the footer. I'll see what I can do

fqueiruga commented:
May I ask why are you not really happy about that Daniel? Maybe something like this?

<img width="1707" alt="Captura de pantalla 2020-07-09 a las 13 48 41" src="https://user-images.githubusercontent.com/5738588/87036492-2f1b1200-c1eb-11ea-9140-f9936833b3b3.png">

<img width="1707" alt="Captura de pantalla 2020-07-09 a las 13 48 51" src="https://user-images.githubusercontent.com/5738588/87036505-32160280-c1eb-11ea-93b3-c56ed1a49737.png">

timja commented:
Just fix the clouds table to not use heading cells in the footer?

fqueiruga commented:
There are many plugins that use a similar pattern: https://github.com/search?q=org%3Ajenkinsci+-path%3Atarget+-repo%3Ajenkinsci%2Fjenkins+sortbottom&type=Code

timja commented:
no reason not to fix it in core imo

fqueiruga commented:
Could be, but I'd rather keep it for another PR

fqueiruga committed:
Set a min-width for the tabs

fqueiruga commented:
I have created some docker images to test versions with separated tabs and grey / white heading combinations:

- `fqueiruga/jenkins-new-ux:tables-tabs-white-header`
- `fqueiruga/jenkins-new-ux:tables-tabs-light-grey-header`

I have found a way to remove background from table footer for the manage nodes and clouds table, works great IMO

josephbrueggen commented:
Thank you @fqueiruga for the explorations, and @timja and @daniel-beck for the feedback! I'm going to review this feedback again, then go back to the design files and see what works best for aligning with our other base styles, and for visual accessibility.

daniel-beck commented:
> May I ask why are you not really happy about that Daniel? Maybe something like this?

Perhaps the use of thickness instead of color/constrast? Unsure.

uhafner commented:
For the tabs I prefer the separate ones that use the thin and light grey lines. I think that the thick and black lines seem to catch the focus too much. 

For the table headers I'm not sure: the light grey headers look good (since they also match the headers of the cards - build widget). But the single line also looks good. But a single line will not look so good if there is no tab used. Then a line before and after the header will make more sense.

fqueiruga commented:
> Perhaps the use of thickness instead of color/constrast? Unsure.

I've always seen this done using thickness and not color. Having a strong color like the primary blue will bring much attention to the line itself. IMO table borders should be as bland as possible. 

> For the tabs I prefer the separate ones that use the thin and light grey lines. I think that the thick and black lines seem to catch the focus too much.

I may agree with that. My problem with separate tabs right now is that I haven't done any field research how separate tabs may impact other uses. We didn't consider that as a possibility. I'd rather push that for a follow up PR.

I'm prefer the light grey headers vs the white ones. I think what you said @uhafner is right, and I think the white header has accessibility issues (difficult to see what's a header and what's not).

IMO some of the reasons why I believe dark header to be a better option it's because it is boring and it fits all cases, even if it's not the prettiest one.

rsandell commented:
Just a bit curious if this will be broken or look off after this change https://github.com/jenkinsci/build-failure-analyzer-plugin/blob/master/src/main/resources/com/sonyericsson/jenkins/plugins/bfa/CauseManagement/index.groovy#L110

fqueiruga commented:
I don't know what effects the cellpadding, spacing and border would have TBH. We aimed to keep the maximum compatibility possible but if some tables use hard inline styling there's not much we can do.

fqueiruga committed:
Fixed mispelled variable

josephbrueggen commented:
Hey everyone, thanks again for the feedback. I think the best course of action within the scope of this PR is to leave the treatment as it was presented in the SIG (see screenshots in original description).

I will investigate these various treatments further throughout the next few weeks to determine how and if they should further change in the short-term. I'll test out the different color treatments, border weights, and tab spacing to see what works best alongside surrounding UI and our other base styles.

In addition to working visually as isolated tab and table treatments, it's important that these styles align with the overall evolution of the UI, so this will take some investigation in the context of the larger design system. I will share the results of looking into this in the next UX SIG meeting, then let's try to introduce any changes in a separate PR.

fqueiruga commented:
Most actual development has been done and the Jira ticket is now properly documented and includes the design documentation. I'm taking it out of draft status, so now I'll address feedback and corrections.

I'd like to know to what extend this design should be changed on this PR or on a follow up revisiting, as @josephbrueggen suggested on https://github.com/jenkinsci/jenkins/pull/4835.

timja commented on war/src/main/less/modules/panes-and-bigtable.less:
```diff
@@ -0,0 +1,142 @@
+  padding: 8px 0px;
+  color: var(--pane-header-color);
+  background: var(--pane-header-bg);
+  // TODO: see how this affects the side panel widgets
```
is this TODO still required?

fqueiruga replied:
Removed

timja commented on war/src/main/less/modules/panes-and-bigtable.less:
```diff
@@ -0,0 +1,142 @@
+}
+
+.bigtable th {
+  font-size: var(--bigtable-header-font-size); // TODO: move this to a <thead> or .header declaration
```
is this TODO still required?

fqueiruga replied:
I'll remove this variable because it's still using 14px font size atm.

timja commented on war/src/main/less/modules/panes-and-bigtable.less:
```diff
@@ -0,0 +1,142 @@
+  display: block;
+  // color: var(--pane-text-color);
+  color: var(--bigtable-header-text-color);
+  font-weight: var(--pane-header-font-weight);  // Should we use the bigtable one here?
```
have you decided which one should be used?

fqueiruga replied:
I have decided on using both. I'll use `--pane-header-color` on `table.sortable` by default but will override it for `bigtable`s. (See https://github.com/jenkinsci/jenkins/pull/4835/commits/4527d84056bd503e226e0f450d2ba6da536cff71)

timja commented on war/src/main/less/modules/panes-and-bigtable.less:
```diff
@@ -0,0 +1,142 @@
+table.sortable a.sortheader {
+  text-decoration: none;
+  display: block;
+  // color: var(--pane-text-color);
```
can we remove commented out code please?

```suggestion
```

timja reviewed (approved):
Tested locally via core-pr tester.
Looks good

Just some minor comments about TODOs

fqueiruga committed:
Update war/src/main/less/modules/panes-and-bigtable.less

Co-authored-by: Tim Jacomb <timjacomb1+github@gmail.com>

fqueiruga committed:
Address review comments

- Have two options for header color in sortable table headers (bigtables or not)
- Remove bigtable header font size variable as it was not changing anything
- Remove outdated comment

uhafner reviewed (commented):
I understand that this design topic is a matter of personal preference. In general I think there are several elements which are not optimal yet (from my point of view):
- the borders are now missing around the builds table
- the build table requires now 50% more height
- the tab lines are too thick
- how do look tabs without tables?
- the color of the header and tab selection is too dark
- there is no separation (space) between tab and table

So in total I think that we should not go live with intermediate changes that will change in a couple of weeks again. This will only cause a lot of confusion and discussion. I think it makes sense to wait for a bigger consensus on that change.

I am not a core developer, so I will not block this PR. But I think we should not hurry and better wait for some more discussions in the UI SIG channel.

fqueiruga commented:
Addressing your points Ulli:


<details>
<summary>the borders are now missing around the builds table</summary>

Only the left & right ones, which doesn't really affect the identification of the table. It's the same as the tables on the warnings-ng pages.
</details>

<details>
<summary>the build table requires now 50% more height</summary>

That's something that has come up in other UI widgets. The Jenkins UI currently needs some whitespacing for proper readability. That said, I guess we could try reducing the vertical cell padding on small icon sizes, I'll try to get back to you on that one.
</details>

<details>
<summary>the tab lines are too think</summary>

Those light grey 1px borders can be difficult to see in low quality / budget monitors.I find that those borders should either be darker or 2px wide.
</details>

<details>
<summary>how do look tabs without tables?</summary>

Imo better than they looked previously. I have also added a `showBaseline` prop and a CSS variable to the `tabBar.jelly` widget to show the line.

<img width="640" alt="Captura de pantalla 2020-07-14 a las 10 29 21" src="https://user-images.githubusercontent.com/5738588/87403381-1630a900-c5bd-11ea-83f7-498e695abc35.png">
<img width="713" alt="Captura de pantalla 2020-07-14 a las 10 29 33" src="https://user-images.githubusercontent.com/5738588/87403386-1761d600-c5bd-11ea-8cda-39210d0924e0.png">
</details>

<details>
<summary>the color of the header and tab selection is too dark</summary>

I think this is the one that's more in the realm of personal preference. I, for example, find lighter headers more inaccessible.

**Idea:** something that I could do immediately is adding style variants to the table so that adding extra CSS classes could modify the header background. WDYT of this?
</details>

<details>
<summary>there is no separation (space) between tab and table</summary>

This is a more difficult one that, if possible, I think could go in it's own PR. I'd like to do some field research to see how the tab bar is used across other plugins. For example, if I add a 1.5rem bottom margin to the tabbar, it may cause it to be too separated in some screens.
</details>

---

If the preferred option is to discuss it on the next SIG meeting I'll do some preparations then, to show a live version of some of the different alternatives then.

fqueiruga committed:
Revert "Unwraps view and plugin tables from .pane-frame" to solve merge conflicts

This reverts commit cfd90c48332a5205471dbe7b53eb37f42772c3fd.

fqueiruga committed:
Merge remote-tracking branch 'upstream/master' into tables-tabs

fqueiruga committed:
Unwraps project & plugin tables from .pane-frame

fqueiruga commented:
@uhafner what would you think on having a `.bigtable--compact` variant that would reduce the vertical paddings? Here's an example of this variant set on the project view when the iconSize is equal to 16x16

<details>
<summary>Icon size S (16x16)</summary>

<img width="1340" alt="Captura de pantalla 2020-07-17 a las 15 34 25" src="https://user-images.githubusercontent.com/5738588/87791867-51dca480-c843-11ea-916e-625c95c437ed.png">
</details>

<details>
<summary>Icon size M</summary>

<img width="1332" alt="Captura de pantalla 2020-07-17 a las 15 34 34" src="https://user-images.githubusercontent.com/5738588/87791860-4ee1b400-c843-11ea-92e5-371c52613bc2.png">
</details>

oleg-nenashev reviewed (approved):
The code looks good to me, but I have not tested it locally.
May get to it tonight, but I do not want to block merge.

We may merge it in 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process

uhafner commented:
> what would you think on having a .bigtable--compact variant that would reduce the vertical paddings? 

Yes, that would help! 

Maybe one addition to my points from above: I think in general it is useful to apply your changes to all those tables in Jenkins (e.g. the padding, styling). I just think that the jobs view is a specific widget that should be handled separately: this widget just uses the same table class by some kind of "coincidence". It would make sense to improve the layout and usability separately (based on the same elements and maybe same classes). E.g., for the build view vertical borders make sense because this widget is used as a kind of portlet (see screenshot). It somehow looks somewhat odd with the other portlets that use borders now. (My tables are all placed into cards, therefore a border is not really required since the card has a border).

timja commented:
Thanks for the reviews all, merging, it can be iterated on.

daniel-beck commented on war/src/main/less/modules/panes-and-bigtable.less:
```diff
@@ -0,0 +1,143 @@
+
+// Keep this for tables still wrapped by .pane-frame
+.pane-frame table, .pane-frame .bigtable tr {
+  border: none; /* Border will be provided by the pane-frame */
```
This rule appears to have caused https://issues.jenkins.io/browse/JENKINS-63677 which makes all progress bars in Jenkins completely useless.
